File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Germany.m
//
//  UIImage+Germany.m
//  CountriesFlags
//

#import "Countries/UIImage+Germany.h"

@implementation UIImage (Germany)

+ (UIImage * _Nullable)flagGermanyWithSize:(CGSize)size {
    UIColor* colorBlack = [UIColor blackColor];
    UIColor* colorRed = [UIColor redColor];
    UIColor* colorGold = [UIColor colorWithRed:1.0f
                                         green:204.0f/255.0f
                                          blue:0.0f
                                         alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorBlack setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorGold setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Germany.swift
//
//  UIImage+Germany.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Germany "wikipedia"
     
     Generates the flag of Germany.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Germany.
     */
    internal static func flagGermany(with size: CGSize) -> UIImage? {
        let colorBlack = UIColor.black
        let colorRed = UIColor.red
        let colorGold = UIColor(red: 1.0,
                                green: 204.0/255.0,
                                blue: 0.0,
                                alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlack.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorRed.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorGold.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
